contributor/04-mechanics-of-contributing.md

```diff
@@ -0,0 +1,125 @@
+
+Great, you've made yourself familiar with the host teams approach and they are looking forward to receive your pull request.
+Which gotchas are there waiting for you now?
+
+First, you'll be in less direct contact with them and second you aren't expected to be as knowledgeable and proficient as you might be on the full-time projects that your team owns.
+How can you now deal with this the best?
+
+Try to peruse their documentation, the conversation archives and code artifacts from the host team to unblock yourself - this is pretty much similar to the situation you and likely most people find yourself in when using one of the popular OSS projects.
+
+However, this time you've got the advantage to be able to directly ask the host team if, even after trying to unblock yourself, things went nowhere. (That's an approach that works with many OSS projects as well!)
+It doesn't do either side any good for you to spin forever on a problem that could be resolved in a few-minute conversation with someone more knowledgeable about the project.
+It's OK to ask for help.
+
+There's one key difference though, bringing advantage to you and other people in the future: 
+In almost all cases you should prefer the projects' official communication channels - this can be a mailing list, a chat room, the issue tracker or something similar depending on the purpose of having a more synchronous or asynchronous way of interacting or the varying needs for structure in the communication.
+All of those usually have in common that they are text-based, archived, searchable and linkable - this means your question and the answer will be written down.
+This way you could benefit from this passively documented knowledge on your search and will help future contributors to have the same advantage.
+Imagine what would be needed for realizing this benefit by just walking by or calling a trusted committer on the project or even in a group call.    
+
+Did you on your way of trying to unblock yourself find that you were missing documentation or found parts of it to be wrong or out-of-date and had to figure this out for yourself now?
+Often projects are happy to receive additions, updates or corrections for their existing documentation as well - you've just found another opportunity to contribute! 
+(Or just politely provide them with feedback on your experience and what would have helped you.)
+
+### Crafting the code
+
+We all have our preferences and opinions on code style, indentation, etc. 
+The home that the host team lives in will have them as well, individual wallpaper, decoration and hangings incl. 
+Try to adapt and match those preferences even if it's not what you would normally do and even if it is not specified in the projects' `CONTRIBUTING.md`.
+If you are unsure you can always ask politely, nevertheless a guest contribution for a feature or bug fix is not the time to introduce a new way of structuring or formatting project code.
+
+## Submitting the pull request
```

Comment:
I'm missing one aspect here: Release early, release often - in order to avoid running in the wrong direction, it can help to submit a draft pull request, ask for early feedback and go from there. There's even a GitHub feature for that.

This advise comes with two caveats:

People in corporate settings usually are unwilling to share non-polished work, exposing themselves to criticism early on. Likely that's particularly true in environments that aren't tolerant wrt. mistakes and weaknesses.

People in corporate settings in a host team may be unwilling to review draft work - and reviewing that over and over until it's done. Most likely this applies to those who are not used to work in tight iteration loops.

Reply:
Important addition!
Especially the case of willingness to share non-polished work or work in the open from very early on due to fear of retaliation or ridicule or exhaustion of help offers or patience.

Not instantly sure how to write that though. Let's see what I can come up with later.

Reply:
Just took a look through the text again and I would think that [this](https://github.com/InnerSourceCommons/InnerSourceLearningPath/pull/117/files#diff-7edd0fadfb4f25c945bdeca937c16632R148-R150) addresses some of this.
I'll add a few more lines to it.